### ISSUE TITLE
Cluster "mixin" should be part of base class

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1065]
There is a `ClusterMixin` in the newly merged clustering module which is quite weird.

The only thing it has is `fit_predict`, but that should be a default functionality in clusterers which do have `fit` and `predict` anyway?

So, let's just move this to the `BaseClusterer`.

**Comment by chrisholder** [comment]
So the original reason ClusterMixin exists is the early sktime version of the Clustering module I made closely mirrored Sklearns where they have the following as their ClusterMixin

```
class ClusterMixin:
    """Mixin class for all cluster estimators in scikit-learn."""

    _estimator_type = "clusterer"

    def fit_predict(self, X, y=None):
        """
        Perform clustering on `X` and returns cluster labels.
        Parameters
        ----------
        X : array-like of shape (n_samples, n_features)
            Input data.
        y : Ignored
            Not used, present for API consistency by convention.
        Returns
        -------
        labels : ndarray of shape (n_samples,), dtype=np.int64
            Cluster labels.
        """
        # non-optimized default implementation; override when a better
        # method is possible for a given clustering algorithm
        self.fit(X)
        return self.labels_

    def _more_tags(self):
        return {"preserves_dtype": []}
```

And then their clustering algorithms inherit ClusterMixin and BaseEstimator like so

```
class DBSCAN(ClusterMixin, BaseEstimator):
    """Perform DBSCAN clustering from vector array or distance matrix.
    DBSCAN - Density-Based Spatial Clustering of Applications with Noise.
    Finds core samples of high density and expands clusters from them.
... etc.
```
Now I am more familiar with Sktimes apporach I agree this mixin is redundent as we inherit BaseEstimator in our Base class (i.e. BaseClusterer, BaseClassifier, BaseForecaster etc.) which as you say does everything the Mixin does. Ill remove this in the next PR.

**Comment by fkiraly** [comment]
>Now I am more familiar with Sktimes apporach I agree this mixin is redundent as we inherit BaseEstimator in our Base class (i.e. BaseClusterer, BaseClassifier, BaseForecaster etc.) which as you say does everything the Mixin does. Ill remove this in the next PR.

I think that's not entirely correct - you shouldn't just remove the mixin, but move the `fit_predict` to the `BaseCluster`.

Btw, should that not be called `BaseClusterer` (or at least `BaseClustererer`)?

**Comment by chrisholder** [comment]
> I think that's not entirely correct - you shouldn't just remove the mixin, but move the `fit_predict` to the `BaseCluster`.

I see what you're saying Ill add that in.

> Btw, should that not be called `BaseClusterer` (or at least `BaseClustererer`)?

Ill leave this up to an executive decision because my reasoning is incredibly superficial I don't like how 'BaseClusterer' looks in my IDE vs 'BaseCluster' (just looks long and something not quite right) so kept it as BaseCluster :joy: .

But yes another ML package like Weka calls it Clusterer as their base class - see https://weka.sourceforge.io/doc.dev/weka/clusterers/Clusterer.html so if people would rather BaseClusterer it can be changed.

**Comment by fkiraly** [comment]
So `BaseClusterererer` it is.

**Comment by fkiraly** [comment]
or `BaseClusteerr`?

**Comment by TonyBagnall** [comment]
BaseClusterer it has to be, if we want to maintain equivalence of syntax. It looks horrible, but the thing that makes clusters is a clusterer

**Comment by fkiraly** [comment]
resolved by #1075
